feat(curriculum): add password generator project to Python curriculum (#52444)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-regular-expressions-by-building-a-password-generator/65649a631440b757f9f7b428.md

---
id: 65649a631440b757f9f7b428
title: Step 28
challengeType: 20
dashedName: step-28
---

# --description--

A regular expression, or regex, is a pattern used to match a specific combination of characters inside a string. It is a valid alternative to `if`/`else` conditional statements when you need to match or find patterns inside a string for validation purposes, character replacement, and others.

The `compile()` function from the `re` module compiles the string passed as the argument into a regular expression object that can be used by other `re` methods.

> Create a new `pattern` variable and give it the value of `re.compile('i')`.

Reviewer: ```suggestion
Declare a new `pattern` variable and assign the value of `re.compile('i')` to this variable.
```
